core/kernel/dt.c

```diff
@@ -566,11 +566,17 @@ void init_external_dt(unsigned long phys_dt)
 		return;
 	}
 
-	fdt = core_mmu_add_mapping(MEM_AREA_EXT_DT, phys_dt, CFG_DTB_MAX_SIZE);
-	if (!fdt)
-		panic("Failed to map external DTB");
-
-	dt->blob = fdt;
+	mtype = core_mmu_get_type_by_pa(phys_dt);
+	if (mtype == MEM_AREA_MAXTYPE) {
+		/* Map the DTB if it is not yet mapped and save it */
+		dt->blob = core_mmu_add_mapping(MEM_AREA_EXT_DT, phys_dt,
+						CFG_DTB_MAX_SIZE);
+		if (!dt->blob)
+			panic("Failed to map external DTB");
+	} else {
+		/* Save the DTB if it is already mapped in one memory area */
```

Comment:
Rather "Save DTB **address**...". Suggestion to fit in 80char/line:
`		/* Get the DTB address if already mapped in a memory area */`

Same comment for line 571 above. It's not the DTB that is saved but its location. I would suggestion to simply remove the 3 last words:
`/* Map the DTB if it is not yet mapped */`

Reply:
Fixed.